fs/zbd_zenfs.cc

```diff
@@ -773,9 +776,10 @@ IOStatus ZonedBlockDevice::AllocateIOZone(Env::WriteLifeTimeHint file_lifetime,
       /* We have to make sure we can open an empty zone */
       while (!got_token && !GetActiveIOZoneTokenIfAvailable()) {
         s = FinishCheapestIOZone();
+        // Note that we may run into a situation that all zones are either empty
+        // or full except busy ones, so we have to keep trying.
         if (!s.ok()) {
-          PutOpenIOZoneToken();
-          return s;
+          Info(logger_, "Finish cheapest zone failed, keep trying...");
```

Comment:
We should probably find a way to check if we fail because all zones are empty or full, or if we are returning because of another error.

Reply:
your are right, maybe let the FinishCheapXx function return success when no zone is ready to finish is better.